extensions/amp-script/0.1/test/integration/test-amp-script.js

```diff
@@ -78,5 +79,47 @@ describe.configure().skipSinglePass().run('amp-script', function() {
         return element.classList.contains('i-amphtml-broken');
       });
     });
+
+    it('should allow without gesture for small size-defined', function*() {
+      yield poll('<amp-script> to be hydrated',
+          () => element.classList.contains('i-amphtml-hydrated'));
+      const impl = element.implementation_;
+
+      // Give event listeners in hydration a moment to attach.
+      yield browser.wait(100);
+
+      sandbox.stub(impl.userActivation_, 'isActive').callsFake(() => false);
+      sandbox.stub(impl, 'getLayout').callsFake(() => Layout.FIXED);
+      sandbox.stub(impl, 'getLayoutBox').callsFake(() => {
+        return {height: 300};
+      });
+      browser.click('button#hello');
+      yield poll('mutations applied', () => {
+        const h1 = doc.querySelector('h1');
+        return h1 && h1.textContent == 'Hello World!';
+      });
+    });
+
+    it('should terminate without gesture for big size-defined', function*() {
+      yield poll('<amp-script> to be hydrated',
+          () => element.classList.contains('i-amphtml-hydrated'));
+      const impl = element.implementation_;
+
+      // Give event listeners in hydration a moment to attach.
+      yield browser.wait(100);
+
+      sandbox.stub(impl.userActivation_, 'isActive').callsFake(() => false);
```

Comment:
Does this default to false? If so, do we need to stub it?

Aside: I really think we should avoid referencing ivars as much as possible in integration tests. IIRC non-hermeticity is one of the main reasons why the UI team is building the new E2E test framework (not to mention issues with single-pass). 

`UserActivation` is special but I think we could add `getMode().test` behavior to simulate the `Event.isTrusted` check.

Reply:
Until I find a better way to test `isTrusted`, I'd rather be explicit here. `getMode().test` doesn't work in the integration part of the test (not 100%, but seems intentional). But in either case, I'd either need to force the `isTrusted` or `!isTrusted` for a full test. For now this will end up the only override. Not fully satisfied by it, but I think this beats the alternatives at this time.